origin-discovery/src/listener/handler_identity.js

```diff
@@ -0,0 +1,117 @@
+const logger = require('./logger')
+
+const { GrowthEventTypes } = require('origin-growth/src/enums')
+const { insertGrowthEvent } = require('./growth')
+
+
+class IdentityEventHandler {
+  /**
+   * Indexes a user in the DB.
+   * @param {Object} user - Origin js user model object.
+   * @param {{blockNumber: number, logIndex: number}} blockInfo
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _indexUser(user, blockInfo) {
+    // TODO(franck): implement me
+    logger.debug(`Indexing user ${user.address} blockInfo ${blockInfo}`)
+    return
+  }
+
+  /**
+   * Records a ProfilePublished event in the growth_event table.
+   * @param {Object} user - Origin js user model object.
+   * @param {{blockNumber: number, logIndex: number}} blockInfo
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _recordGrowthProfileEvent(user, blockInfo) {
+    // Check profile is populated.
+    const profile = user.profile
+    const validProfile = (profile.firstName.length > 0 || profile.lastName.length > 0) &&
+      (profile.avatar.length > 0)
+    if (!validProfile) {
+      return
+    }
+
+    // Record the event.
+    await insertGrowthEvent(
+      user.address,
+      GrowthEventTypes.ProfilePublished,
+      null,
+      { blockInfo }
+    )
+
+  }
+
+  /**
+   * Records AttestationPublished events in the growth_event table.
+   * @param {Object} user - Origin js user model object.
+   * @param {{blockNumber: number, logIndex: number}} blockInfo
+   * @returns {Promise<void>}
+   * @private
+   */
+  async _recordGrowthAttestationEvents(user, blockInfo) {
+    const topicToEventType = {
+      'facebook': GrowthEventTypes.FacebookAttestationPublished,
+      'airbnb': GrowthEventTypes.AirbnbAttestationPublished,
+      'twitter': GrowthEventTypes.TwitterAttestationPublished,
+      'phone': GrowthEventTypes.PhoneAttestationPublished
```

Comment:
Should we also include the `email` attestation here?

Reply:
Yes ! good catch